classes/teacher.rb

```diff
@@ -0,0 +1,15 @@
+require './classes/person'
+
+# Define a sub-base class
+class Teacher < Person
+  def initialize(specialization, age)
+    super(age)
```

Comment:
- [ ] Kindly make sure to call the `super` method correctly with the right parameters according to the parent's constructor. The `initialize` method here should accept all of the parameters to pass down to the `super` constructor.

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
```